tests/codeception/functional/Content/StanfordNewsCest.php

```diff
@@ -115,11 +115,105 @@ public function testDefaultVariantHidesFields(FunctionalTester $I) {
     $I->canSeeInField('Headline', $default_news->label());
     $I->canSeeInField('Dek', $default_news->get('su_news_dek')->value);
     $I->canSeeInField('Byline', $default_news->get('su_news_byline')->value);
-    
+
     $I->amOnPage($spotlight_news->toUrl('edit-form')->toString());
     $I->canSeeInField('Headline', $spotlight_news->label());
-    // $I->selectOption('Layout', 'Spotlight'); 
+    // $I->selectOption('Layout', 'Spotlight');
     $I->canSeeInField('Quote / Big Text', $spotlight_news->get('su_news_quote')->value);
     $I->canSeeInField('Subtitle', $spotlight_news->get('su_news_subtitle')->value);
   }
+
+  /**
+   * Test Related Spotlights view configuration and existence.
+   */
+  #[CodeceptionAttribute\Group('news_variant')]
+  public function testRelatedSpotlightsViewExists(FunctionalTester $I) {
+    // Log in as administrator to access views admin.
+    $I->logInWithRole('administrator');
+
+    // Verify the view display exists and is configured properly.
+    $I->amOnPage('/admin/structure/views/view/stanford_news/edit/related_spotlights');
+    $I->canSee('Related Spotlights');
+  }
+
+  /**
+   * Test that Related Spotlights filters by matching taxonomy terms.
+   *
+   * This tests filtering via view configuration (arguments and filters).
+   */
+  #[CodeceptionAttribute\Group('news_variant')]
+  public function testRelatedSpotlightsFiltersByTaxonomy(FunctionalTester $I) {
+
+    // Create taxonomy terms.
+    $term_a = $I->createEntity([
+      'name' => 'Category A',
```

Comment:
Make the name dynamic using the faker service. You don't even have to use words.. just use the `$this->faker->uuid()` and then the contextual argument will be `$term_a->label()`

This goes for all entities below.